lib/cisco_node_utils/yum.rb

```diff
@@ -46,11 +61,19 @@ def self.validate(pkg)
         if pkg_info[3].nil?
           query_name = pkg_info[1]
         else
-          query_name = "#{pkg_info[1]}.#{pkg_info[3]}"
+          if platform == :nexus
+            query_name = "#{pkg_info[1]}.#{pkg_info[3]}"
+          elsif platform == :ios_xr
+            # ex query_name xrv9k-k9sec-1.0.0.0-r61102I.x86_64
+            query_name = \
+              "#{pkg_info[1]}-#{pkg_info[2]}-#{pkg_info[3]}.#{pkg_info[4]}"
+          end
         end
       end
       should_ver = pkg_info[2] if pkg_info && pkg_info[3]
       ver = query(query_name)
+      debug "Installed package version #{ver}, expected package version" \
```

Comment:
I just followed the existing code to print debug messages. I can change it to Cisco::Logger.debug.